enumerables.rb

```diff
@@ -0,0 +1,66 @@
+# enumerable.rb
+module Enumerable
+  def my_each
+    i = 0
+    while to_a.length > i
+      yield(to_a[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while to_a.length > i
+      yield(to_a[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    to_a.my_each { |item| new_array.push(item) if yield(item) }
+    new_array
+  end
+
+  def my_all?
+    to_a.my_each { |item| return false if yield(item) == false }
+    true
+  end
+
+  def my_any?
+    to_a.my_each { |item| return true if yield(item) }
+    false
+  end
```

Comment:
1. enumerables #my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
Failure/Error: expect(true_array.my_any?).to be true_array.any?
2. enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true
 Failure/Error: expect(false_array.my_any?).to be false_array.any?
3. enumerables #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric
Failure/Error: expect(array.my_any?(Numeric)).to be array.any?(Numeric)
4. enumerables #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer
Failure/Error: expect(words.my_any?(Integer)).to be words.any?(Integer)
5.  enumerables #my_any? when a Regex is passed as an argument returns true if any of the collection matches the Regex
 Failure/Error: expect(words.my_any?(/d/)).to be words.any?(/d/)
6. enumerables #my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
Failure/Error: expect(words.my_any?(/z/)).to be words.any?(/z/)
7. enumerables #my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
8.  enumerables #my_any? when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern
Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')

Reply:
Solved all the requested changes and also tested it